pex/pex_rules.bzl

```diff
@@ -82,6 +83,16 @@ def _collect_transitive_reqs(ctx):
   return transitive_reqs
 
 
+def _collect_repos(ctx):
+  repos = {}
+  for dep in ctx.attr.deps:
+    if hasattr(dep.py, "repos"):
+      repos += dep.py.repos
+  for file in egg_file_types.filter(ctx.files.repos):
```

Comment:
@benley added support for source distributions as well as cache disabling.

Have some mixed feelings about supporting URLs for the same reason that it's not exactly great practice. I'll think on the issue a bit this week.